src/framegrab/grabber.py

```diff
@@ -512,36 +512,36 @@ def __init__(self, config: dict):
 
         # Assign camera based on serial number if 1) serial_number was provided and 2) we know the
         # serial numbers of plugged in devices
-        if serial_number and found_cams:
+        if found_cams:
+            logger.debug(f"Found {len(found_cams)} USB cameras with Linux commands. Assigning camera by serial number.")
             for found_cam in found_cams:
-                if serial_number != found_cam["serial_number"]:
+                if serial_number and serial_number != found_cam["serial_number"]:
                     continue
 
                 idx = found_cam["idx"]
                 if idx in GenericUSBFrameGrabber.indices_in_use:
-                    raise ValueError(
-                        f"USB camera index {idx} already in use. "
-                        f"Did you use the same serial number ({serial_number}) for two different cameras?"
-                    )
+                    continue
 
-                capture = cv2.VideoCapture(idx)
-                if capture.isOpened():
-                    break  # Found a valid capture, no need to look any further
+                capture = self._connect_and_validate_capture(found_cam)
+                if capture is not None:
+                    break  # Found a valid capture
 
             else:
```

Comment:
Never knew about for else, neat

Reply:
Some people hate it, but I find it useful in certain cases.